packages/compass-aggregations/src/modules/workspace.ts

```diff
@@ -30,9 +33,17 @@ const reducer: Reducer<State, Actions | AggregationActions> = (state = INITIAL_S
   }
 };
 
-export const changeWorkspace = (view: Workspace): WorkspaceChangedAction => ({
-  type: ActionTypes.WorkspaceChanged,
-  view,
-});
-
+export const changeWorkspace = (view: Workspace): ThunkAction<void, RootState, void, Actions> => {
+  return (dispatch) => {
+    // As user switches to builder view, we cancel running ops
+    if (view === 'builder') {
+      dispatch(cancelAggregation());
+      dispatch(cancelCount());
+    }
```

Comment:
We're definitely getting there. I think you would also want to do this when Run button is pressed if there is an in-flight request running, otherwise I can overload server pretty easily (but more importantly if I'm running a log query and maybe change collation or maxtimems there is a chance of running into race condition issue):

![Kapture 2022-04-25 at 17 50 21](https://user-images.githubusercontent.com/5036933/165126385-a26386ed-6b76-41e9-aa24-a7db40f2d359.gif)